app/pods/agendas/route.js

```diff
@@ -3,15 +3,74 @@ import { inject as service } from '@ember/service';
 import { action } from '@ember/object';
 
 export default class AgendasRoute extends Route {
+  @service store;
   @service router;
   @service('session') simpleAuthSession;
 
+  dateRegex = /^(?:(\d{1,2})-)??(?:(\d{1,2})-)?(\d{4})$/;
+
+  queryParams = {
+    filter: {
+      refreshModel: true,
+    },
+    page: {
+      refreshModel: true,
+    },
+    size: {
+      refreshModel: true,
+    },
+    sort: {
+      refreshModel: true,
+    },
+  };
+
   beforeModel(transition) {
     this.simpleAuthSession.requireAuthentication(transition, 'login');
   }
 
-  redirect() {
-    this.transitionTo('agendas.overview');
+  async model(params) {
+    const queryParams = {
+      sort: params.sort,
+      page: {
+        number: params.page,
+        size: params.size,
+      },
+    };
+
+    if (params.filter) {
+      const date = params.filter.split('/').join('-');
+      const match = this.dateRegex.exec(date);
+
+      if (match) {
+        const [, day, month, year] = match.map(num => parseInt(num, 10));
+
+        const from = new Date(year, (month - 1) || 0, day || 0);
+        const to = new Date(from);
+        if (day) {
+          to.setDate(day + 1);
+        } else if (month) {
+          to.setMonth(month); // months are 0-indexed, no +1 required
+        } else {
+          to.setYear(year + 1);
+        }
+
+        queryParams['filter[created-for][:gte:planned-start]'] = from.toISOString();
+        queryParams['filter[created-for][:lte:planned-start]'] = to.toISOString();
+      }
+    }
+
+    return await this.store.query('agenda', queryParams);
```

Comment:
You are querying all agendas resulting in meetings with multiple agendas to be in your set multiple times.
You probably want to filter on the highest Version by filtering on `has-no:nextVersion` or something.
Or go back to queries on `meeting`, but that's probably not desired.

![image](https://user-images.githubusercontent.com/22245223/164028350-ffc42d41-fa00-4b24-85c0-1750a277c780.png)